src/magic_spell_effect.cpp

```diff
@@ -503,3 +509,245 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
+    }
+}
+
+void spell_effect::fatigue( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    caster.add_msg_if_player( m_warning, _( "The fabric of space seems to decay." ) );
+    int x = rng( target.x - 3, target.x + 3 ), y = rng( target.y - 3, target.y + 3 );
+    sp.create_field( {x, y, target.z} );
+}
+
+void spell_effect::pulse( const spell &sp, const tripoint &target )
+{
+    sounds::sound( target, 30, sounds::sound_t::combat, _( "The earth shakes!" ), true, "misc",
+                   "earthquake" );
+    for( const tripoint &pt : g->m.points_in_radius( target, sp.aoe() ) ) {
+        g->m.bash( pt, sp.damage() );
+        g->m.bash( pt, sp.damage() );  // Multibash effect, so that doors &c will fall
+        g->m.bash( pt, sp.damage() );
+        if( g->m.is_bashable( pt ) && rng( 1, 10 ) >= 3 ) {
+            g->m.bash( pt, 999, false, true );
+        }
+    }
+}
+
+void spell_effect::entrance( const spell &sp, const tripoint &target )
+{
+    for( const tripoint &dest : g->m.points_in_radius( target, sp.aoe() ) ) {
+        monster *const mon = g->critter_at<monster>( dest, true );
+        if( mon && mon->friendly == 0 && rng( 0, 600 ) > mon->get_hp() ) {
+            mon->make_friendly();
+        }
+    }
+}
+
+void spell_effect::bugs( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    int roll = rng( 1, 10 );
+    mtype_id bug = mtype_id::NULL_ID();
+    int num = 0;
+    std::vector<tripoint> empty;
+    for( const tripoint &dest : g->m.points_in_radius( target, sp.aoe() ) ) {
+        if( g->is_empty( dest ) ) {
+            empty.push_back( dest );
+        }
+    }
+    if( empty.empty() || roll <= 4 ) {
+        caster.add_msg_if_player( m_warning, _( "Flies buzz around you." ) );
+    } else if( roll <= 7 ) {
+        caster.add_msg_if_player( m_warning, _( "Giant flies appear!" ) );
+        bug = static_cast<mtype_id>( "mon_fly" );
+        num = rng( 2, 4 );
+    } else if( roll <= 9 ) {
+        caster.add_msg_if_player( m_warning, _( "Giant bees appear!" ) );
+        bug = static_cast<mtype_id>( "mon_bee" );
+        num = rng( 1, 3 );
+    } else {
+        caster.add_msg_if_player( m_warning, _( "Giant wasps appear!" ) );
+        bug = static_cast<mtype_id>( "mon_wasp" );
+        num = rng( 1, 2 );
+    }
+    if( bug ) {
+        for( int j = 0; j < num && !empty.empty(); j++ ) {
+            const tripoint spawnp = random_entry_removed( empty );
+            if( monster *const b = g->summon_mon( bug, spawnp ) ) {
+                b->friendly = -1;
+                b->add_effect( static_cast<efftype_id>( "pet" ), 1_turns, num_bp, true );
+            }
+        }
+    }
+}
+
+void spell_effect::light( const spell &sp, const Creature &caster )
+{
+    caster.add_msg_if_player( _( "The %s glows brightly!" ), sp.get_obj_name() );
+    g->events.add( EVENT_ARTIFACT_LIGHT, calendar::turn + 3_minutes );
+}
+
+void spell_effect::growth( const tripoint &target )
```

Comment:
i would much prefer this be done through the infrastructure introduced in #32329